packages/snack-sdk/src/transports/TransportImplSocketIO.ts

```diff
@@ -0,0 +1,104 @@
+import { io } from 'socket.io-client';
+import type { Socket } from 'socket.io-client';
+
+import { ProtocolOutgoingMessage, ProtocolIncomingMessage, ProtocolCodeMessage } from './Protocol';
+import TransportImplBase from './TransportImplBase';
+import type { SnackTransportOptions } from './types';
+
+interface ServerToClientEvents {
+  message: (data: { channel: string; message: ProtocolIncomingMessage; sender: string }) => void;
+  joinChannel: (data: { channel: string; sender: string }) => void;
+  leaveChannel: (data: { channel: string; sender: string }) => void;
+}
+
+interface ClientToServerEvents {
+  message: (data: { channel: string; message: ProtocolOutgoingMessage; sender: string }) => void;
+  subscribeChannel: (data: { channel: string; sender: string }) => void;
+  unsubscribeChannel: (data: { channel: string; sender: string }) => void;
+}
+
+export default class TransportImplSocketIO extends TransportImplBase {
+  private readonly _snackpubURL: string;
+  private _socket: Socket<ServerToClientEvents, ClientToServerEvents> | null = null;
```

Comment:
Another personal preference, feel free to ignore. But with TypeScript, we can already define `private`/`public`/`protected`, we don't need to do that in the name anymore :)

```suggestion
  private readonly snackpubURL: string;
  private socket: Socket<ServerToClientEvents, ClientToServerEvents> | null = null;
```